### PR TITLE
Add gentec security center

### PR DESCRIPTION
**Pull request by YuvHayun**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [CIAC-9554](https://jira-dc.paloaltonetworks.com/browse/CIAC-9554)

### CODE REVIEW

File: Packs/GenetecSecurityCenter/Integrations/GenetecSecurityCenterEventCollector/GenetecSecurityCenterEventCollector.py [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+        super().__init__(base_url=base_url, verify=verify, proxy=proxy, auth=auth, headers={"Accept": "text/json"})
+
+    def _encode_authorization(self, username: str, password: str, app_id: str):
+        updated_username = f"{username};{app_id}"
+        return HTTPBasicAuth(updated_username, password)
+
+    def http_request(self, url_suffix):
+        response_audit = self._http_request('GET', url_suffix=url_suffix, resp_type='response')
+        response = json.loads(response_audit.content)["Rsp"]
+        if response["Status"] == "Fail":
```
Comment (JudahSchwartz) [inline Packs/GenetecSecurityCenter/Integrations/GenetecSecurityCenterEventCollector/GenetecSecurityCenterEventCollector.py:54]: add log

File: Packs/GenetecSecurityCenter/Integrations/GenetecSecurityCenterEventCollector/GenetecSecurityCenterEventCollector.py [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+    start_time: datetime = time_now - timedelta(minutes=1)
+    time_now_str = time_now.strftime(DATE_FORMAT_EVENT)
+    start_time_str = start_time.strftime(DATE_FORMAT_EVENT)
+    query = f"TimeRange.SetTimeRange({start_time_str},{time_now_str})"
+    url_suffix = f"{AUDIT_TRAIL_ENDPOINT}?q={query}"
+    client.http_request(url_suffix=url_suffix)
+    return "ok"
+
+
+def fetch_events_command(
```
Comment (JudahSchwartz) [inline Packs/GenetecSecurityCenter/Integrations/GenetecSecurityCenterEventCollector/GenetecSecurityCenterEventCollector.py:76]: add logs throughout

File: Packs/GenetecSecurityCenter/Integrations/GenetecSecurityCenterEventCollector/GenetecSecurityCenterEventCollector.py [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+                CommandResults(readable_output=tableToMarkdown("Events:", events, headers=GET_EVENTS_HEADERS))
+            )
+
+        elif command == "fetch-events":
+            should_push_events = True
+            last_run = demisto.getLastRun()
+            events = fetch_events_command(
+                client, params, last_run=last_run
+            )
+            last_run_time = datetime.strptime(events[-1]["ModificationTimeStamp"],
```
Comment (JudahSchwartz) [inline Packs/GenetecSecurityCenter/Integrations/GenetecSecurityCenterEventCollector/GenetecSecurityCenterEventCollector.py:154]: double check on deduping the last events with the latest time

File: Packs/GenetecSecurityCenter/Integrations/GenetecSecurityCenterEventCollector/GenetecSecurityCenterEventCollector.py [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+            max_fetch=max_fetch,
+            app_id=app_id,
+            verify=verify,
+            proxy=proxy,
+        )
+
+        if command == "test-module":
+            return_results(test_module(client))
+
+        elif command == "genetec-security-center-get-events":
```
Comment (JudahSchwartz) [inline Packs/GenetecSecurityCenter/Integrations/GenetecSecurityCenterEventCollector/GenetecSecurityCenterEventCollector.py:142]: lets add time to fetch from as an argument here

File: Packs/GenetecSecurityCenter/Integrations/GenetecSecurityCenterEventCollector/GenetecSecurityCenterEventCollector.py [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+from CommonServerPython import *  # noqa # pylint: disable=unused-wildcard-import
+from CommonServerUserPython import *  # noqa
+from requests.auth import HTTPBasicAuth
+import json
+from xmltodict import parse
+
+disable_warnings()
+
+""" CONSTANTS """
+
```
Comment (JudahSchwartz) [inline Packs/GenetecSecurityCenter/Integrations/GenetecSecurityCenterEventCollector/GenetecSecurityCenterEventCollector.py:12]: add at least 1 json file under test_data that has a raw response that can be used as an example

**Review by JudahSchwartz — COMMENTED** [review]
looks great! a couple comments

File: Packs/GenetecSecurityCenter/Integrations/GenetecSecurityCenterEventCollector/GenetecSecurityCenterEventCollector.py (hunk 1 above)
Comment (YuvHayun) [reply Packs/GenetecSecurityCenter/Integrations/GenetecSecurityCenterEventCollector/GenetecSecurityCenterEventCollector.py:54]: The raised exception should print the information to the integration configuration screen, is it still required?

File: Packs/GenetecSecurityCenter/Integrations/GenetecSecurityCenterEventCollector/GenetecSecurityCenterEventCollector.py (hunk 2 above)
Comment (YuvHayun) [reply Packs/GenetecSecurityCenter/Integrations/GenetecSecurityCenterEventCollector/GenetecSecurityCenterEventCollector.py:76]: I have logs for the number of fetched events, the new last_run and the query.. are there any other logs type required?

File: Packs/GenetecSecurityCenter/Integrations/GenetecSecurityCenterEventCollector/GenetecSecurityCenterEventCollector.yml [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+  name: url
+  required: true
+  type: 0
+  section: Connect
+- name: credentials
+  display: Username
+  required: true
+  defaultvalue:
+  type: 9
+  additionalinfo: Username & Password.
```
Comment (ShirleyDenkberg) [inline Packs/GenetecSecurityCenter/Integrations/GenetecSecurityCenterEventCollector/GenetecSecurityCenterEventCollector.yml:16]:
```suggestion
  additionalinfo: Username and password.
```

File: Packs/GenetecSecurityCenter/Integrations/GenetecSecurityCenterEventCollector/GenetecSecurityCenterEventCollector.yml [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+- display: Use system proxy settings
+  name: proxy
+  type: 8
+  required: false
+description: Genetec Security Center Audit Trail Event Collector.
+display: Genetec Security Center Event Collector
+name: Genetec Security Center Event Collector
+script:
+  commands:
+  - description: Manual execution of fetch-events, use this for debugging purposes.'
```
Comment (ShirleyDenkberg) [inline Packs/GenetecSecurityCenter/Integrations/GenetecSecurityCenterEventCollector/GenetecSecurityCenterEventCollector.yml:43]:
```suggestion
  - description: Manual execution of fetch-events. Use this for debugging purposes.
```

File: Packs/GenetecSecurityCenter/Integrations/GenetecSecurityCenterEventCollector/README.md [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+## Configure Armis Event Collector on Cortex XSOAR
+
+1. Navigate to **Settings** > **Integrations** > **Servers & Services**.
+2. Search for Genetec Security Center Event Collector.
+3. Click **Add instance** to create and configure a new integration instance.
+
+    | **Parameter** | **Description** | **Required** |
+    | --- | --- | --- |
+    | Your server URL |  | True |
+    | Username | Username &amp;amp; Password. | True |
```
Comment (ShirleyDenkberg) [inline Packs/GenetecSecurityCenter/Integrations/GenetecSecurityCenterEventCollector/README.md:12]:
```suggestion
    | Username | Username and password. | True |
```

File: Packs/GenetecSecurityCenter/Integrations/GenetecSecurityCenterEventCollector/README.md [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+    | Password |  | True |
+    | Application ID |  | True |
+    | Trust any certificate (not secure) |  | False |
+    | Use system proxy settings |  | False |
+    | Maximum number of events per fetch | Alerts and activity events. |  |
+
+4. Click **Test** to validate the URLs, token, and connection.
+
+## Commands
+You can execute these commands Alert War Room in the CLI in XSIAM.
```
Comment (ShirleyDenkberg) [inline Packs/GenetecSecurityCenter/Integrations/GenetecSecurityCenterEventCollector/README.md:22]:
```suggestion
You can execute these commands in the War Room in the CLI in Cortex XSIAM.
```

**Comment by ShirleyDenkberg** [comment]
@JudahSchwartz Doc review completed.

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py [hunk 10]
```diff
@@ -11695,6 +11700,9 @@ def send_data_to_xsiam(data, vendor, product, data_format=None, url_key='url', n
     :param should_update_health_module: whether to trigger the health module showing how many events were sent to xsiam
         This can be useful when using send_data_to_xsiam in batches for the same fetch.
 
+    :type add_proxy_to_request ``bool``
```
Comment (JudahSchwartz) [inline Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:11703]:
```suggestion
    :type add_proxy_to_request: ``bool``
```

**Review by JudahSchwartz — APPROVED** [review]

**Pull request merged**